### PR TITLE
Closes #75

### LINKED ISSUE #75: Add a list of functions
In `01-...`, the "Calculated values" header mentions functions. Maybe a table would be nice.

### PR DESCRIPTION
**Pull request by waveform80**
Adds a section just after joins detailing scalar SQL functions (with particular emphasis on COALESCE & NULLIF and their use in JOINs).

### CODE REVIEW

File: 03-sql-joins-aliases.md [hunk 1]
```diff
@@ -69,6 +69,83 @@ could do something like
 > Write a query that finds the average weight of each rodent species (i.e., only include species with Rodent in the taxa field).
 
 
+## Functions
+
+SQL includes numerous functions for manipulating data. You've already seen some
+of these being used for aggregation (SUM and COUNT) but there are functions
```
Comment (tpoisot) [inline 03-sql-joins-aliases.md]: Please use backticks around keywords, etc. COALESCE -> `COALESCE`

Comment (waveform80) [inline 03-sql-joins-aliases.md]: Updated the commit to backtick quote all keywords

**Comment by tpoisot** [comment]
@orchid00 -- your advice?

**Comment by waveform80** [comment]
Doh - just noticed I screwed up a couple of quotes there. Should be fixed now.

**Comment by orchid00** [comment]
.... ahm, honestly I'm not sure if this is necessary for beginners. I personally think that this will be a bit confusing, and I'll leave it out. But, if you guys think it should be added, then IFNULL() does the same as COALESCE(), (I think IFNULL() is more intuitive to remember?) but also depending on the DBMS people can also, ask or know about ISNULL() (which will also be more complicated to explain to novices)

**Comment by waveform80** [comment]
IFNULL is indeed more intuitive, but I'm afraid the standard is COALESCE (the former is supported by SQLite and I'm happy to update the PR if that's preferred, but I generally encourage the use of COALESCE simply because it's universally supported - one of the rare cases of all the engines actually agreeing on something :-). It's also worth noting that IFNULL is two argument only, while COALESCE is multi-argument (although admittedly use of more than two arguments is extremely rare - I think I've used it once in a decade!).

On the complexity issue - is it that functions generally are going to be too complex to cover (ie should #75 be closed), or is it that I've picked particularly hard functions (ie a different PR with a different selection of functions would be acceptable)?

**Comment by tpoisot** [comment]
I'll side with @orchid00 -- as we discussed in other issues, this is primarily a SQLite lesson. Let's go with *intuitive* first.

@waveform80 please add a new commit so we can see what changed.

**Comment by waveform80** [comment]
Sorry - just to be clear (given that most projects I work on expect rebasing for single-commit merges) - you'd prefer I use multiple commits in the PR rather than re-writing the existing commit?

**Comment by tpoisot** [comment]
Yes.

Sent from a mobile device. Pardon the typos/brevity. On Fri, Apr 01, 2016 at 9:05 AM, Dave Jones < notifications@github.com [notifications@github.com] > wrote:
Sorry - just to be clear (given that most projects I work on expect rebasing for single-commit merges) - you'd prefer I use multiple commits in the PR rather than re-writing the existing commit?

**Comment by waveform80** [comment]
Okay, no prob - extra commit is above (I figured I'd add it anyway as I could always rebase to squash 'em together)

**Comment by tpoisot** [comment]
@waveform80 cool - github now squashes upon merge, so having different commits is not a problem.

**Comment by waveform80** [comment]
Yeah - just noticed that new drop-down while working on another project today - nice stuff!

**Pull request merged**